### PR TITLE
search between dates, extend information

### PR DESCRIPTION
**Pull request by tomahock**
#23 issue

@josepostiga can you check Incident Model, to set the correct types?

### CODE REVIEW

File: app/Http/Controllers/IncidentController.php [hunk 1]
```diff
@@ -33,25 +33,54 @@ public function active(Request $request): JsonResponse
...
+        $after = $request->get('after');
+
+        if($day && ($before || $after)){
+            abort(422);
+        }
+
+        $timeRange = false;
+        if($after){
+            $after = new Carbon($after);
+            $before = $before ? new Carbon($before) : new Carbon();
```
Comment (josepostiga) [inline app/Http/Controllers/IncidentController.php:46]: We can simplify, here, a little:
```suggestion
            $before = new Carbon($before)
```

`$before` will be initialized either with a value or with null, due to how `$request->get()` works.

File: app/Http/Controllers/IncidentController.php [hunk 2]
```diff
@@ -33,25 +33,54 @@ public function active(Request $request): JsonResponse
...
         $all = $request->get('all');
+        $extend = $request->get('extend');
         $concelho = $request->get('concelho');
 
-        $incidents = Incident::whereBetween(
+
+        $incidents = Incident::when($day, function($query, $day){
+            return $query->whereBetween(
+                'dateTime',
+
```
Comment (josepostiga) [inline app/Http/Controllers/IncidentController.php:62]: Nit: extra space, here.
```suggestion
```

File: app/Http/Controllers/IncidentController.php [hunk 3]
```diff
@@ -33,25 +33,54 @@ public function active(Request $request): JsonResponse
...
+                ]
+            );
+        })->when($timeRange, function($query,$timeRange){
+            return $query->whereBetween(
                 'dateTime',
                 [
-                    Carbon::parse($date->startOfDay()),
-                    Carbon::parse($date->endOfDay())
+                    Carbon::parse($timeRange[0]->startOfDay()),
+                    Carbon::parse($timeRange[1]->endOfDay())
```
Comment (josepostiga) [inline app/Http/Controllers/IncidentController.php:73]: According to the Carbon docs, the `startOfDay()` and `endOfDay()` methods returns new Carbon instances. I think the parse() call is not required, right?

Comment (tomahock) [reply app/Http/Controllers/IncidentController.php:73]: Only work this way. If i remove the `parse()` it doesn't work.

**Pull request merged**